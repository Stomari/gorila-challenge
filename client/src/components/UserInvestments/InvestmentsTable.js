import React from 'react';

const InvestmentsTable = ({ investmentInfo, deleteInvestment }) => {
  return (
    <table>
      <thead>
        <tr>
          <th>Data da compra</th>
          <th>Valor</th>
        </tr>
      </thead>
      <tbody>
        {
          investmentInfo && investmentInfo.map((element) => {
            return (
              <tr key={element._id}>
                <td>{element.date}</td>
                <td>{element.value}</td>
                <td><img className="delete" onClick={() => deleteInvestment(element._id)} src="./images/trash.png" alt="Trash Icon" /></td>
              </tr>
            )
          })
        }
      </tbody>
    </table>
  )
}

export default InvestmentsTable;