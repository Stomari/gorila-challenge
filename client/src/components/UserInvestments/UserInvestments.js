import React, { useState } from 'react';
import InvestmentTable from './InvestmentsTable';

const UserInvestments = ({ fixedIncome, variableIncome, deleteInvestment }) => {
  const [showFixed, setShowFixed] = useState(false);
  const [showVariable, setShowVariable] = useState(false);

  return (
    <section className="investments">
      <div className="invest-table">
        <div className="table-toggle fixed-title" onClick={() => setShowFixed(!showFixed)}>
          <h2>Renda Fixa:</h2>
          {
            showFixed ?
              <p>Clique para esconder!</p>
              :
              <p>Clique para mostrar!</p>
          }
        </div>
        {
          showFixed &&
          <InvestmentTable investmentInfo={fixedIncome} deleteInvestment={deleteInvestment} />
        }
      </div>
      <div className="invest-table">
        <div className="table-toggle variable-title" onClick={() => setShowVariable(!showVariable)}>
          <h2>Renda Vari√°vel:</h2>
          {
            showVariable ?
              <p>Clique para esconder!</p>
              :
              <p>Clique para mostrar!</p>
          }
        </div>
        {
          showVariable &&
          <InvestmentTable investmentInfo={variableIncome} deleteInvestment={deleteInvestment} />
        }
      </div>
    </section>
  )
}

export default UserInvestments;