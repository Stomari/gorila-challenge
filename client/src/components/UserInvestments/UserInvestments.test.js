import React from 'react';
import { shallow } from 'enzyme';
import UserInvestments from './UserInvestments';

describe('UserInvestments rendering', () => {
  const component = shallow(<UserInvestments />); 

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render investments', () => {
    expect(component.find('.investments').length).toEqual(1);
  });

  it('should render fixed and variable investments table', () => {
    expect(component.find('.invest-table').length).toEqual(2);
  });
});