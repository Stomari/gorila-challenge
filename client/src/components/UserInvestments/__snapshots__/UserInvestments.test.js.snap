// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInvestments rendering should render 1`] = `ShallowWrapper {}`;
